pkg/clients/aws.go

```diff
@@ -102,12 +105,35 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err := c.Get(ctx, types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}, s); err != nil {
 			return nil, errors.Wrap(err, "cannot get credentials secret")
 		}
-		return UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		cfg, err := UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		return SetResolver(ctx, mg, cfg), err
 	default:
 		return nil, errors.Errorf("credentials source %s is not currently supported", s)
 	}
 }
 
+// SetResolver parses annotations from the managed resource
+// and returns a configuration accordingly.
+func SetResolver(ctx context.Context, mg resource.Managed, cfg *aws.Config) *aws.Config {
+	if URL, ok := mg.GetAnnotations()["aws.crossplane.io/endpointURL"]; ok {
```

Comment:
Do these endpoints change on a service-to-service basis? I'm wondering if we should put the annotations on the `ProviderConfig` instead (or even make them a first class field). If we overrode the endpoint on the `ProviderConfig`, would the same endpoint apply to both RDS instances and S3 buckets (for example)?

Reply:
it changes from service to service unfortunately